### DISCUSSION TITLE
Fetch error "getaddrinfo ENOTFOUND" in production only

### DISCUSSION THREAD

**Discussion by jimburch** [vercel/vercel#8800] [Help]
I'm getting a build error happening in production only, this code runs locally without issue:

Front end component:
```js
const Home = ({ newTake }) => {
  const [take, setTake] = useState(newTake);
  const [loading, setLoading] = useState(false);

  return (
    <TakeContext.Provider value={{ take, setTake }}>
      <div className={styles.root}>
        <Header />
        {loading ? (
          <div
            className={styles.ldsHourglass}
            style={{ margin: "50px 0px 50px 0px" }}
          />
        ) : (
          <Tweet />
        )}
        {/* <NewTake generateTake={generateTake} /> */}
      </div>
    </TakeContext.Provider>
  );
};
```

Data fetching component:
```js
export const getStaticProps = async () => {
  const response = await fetch(`${API_URL}/hoopsbot/random`, {
    headers: {
      Authorization: `bearer ${API_BEARER_TOKEN}`,
    },
  })
    .then((response) => response.json())
    .catch((error) => {
      throw new Error(error);
    });

  return {
    props: {
      newTake: response,
    },
  };
};
```

Locally, this returns a json body just fine (where the API URL is my localhost) but I get the following build error in production (API url omitted):
```
Error: FetchError: request to https://api.************* failed, reason: getaddrinfo ENOTFOUND api.***********
```

I've tried all sorts of variations of data fetching, using the `_api` folder, `getServerSideProps`, using axios instead of fetch, but nothing works on production.

**Comment by amyegan** [comment]
Hi @JimBurch! Is the `API_URL` going to something hosted externally, or is it part of the app you're trying to build? Sometimes there can be problems with accessing API routes in `getStaticProps` since the whole project is still being built.

Have you double checked that you have valid `API_URL` and `API_BEARER_TOKEN` [environment variables](https://vercel.com/docs/concepts/projects/environment-variables) available in the production environment? Differences in env vars between development and production environments are a common cause of connection problems. This isn't usually the error we see for that problem, though it's still possible.

Or maybe it's an `HTTP` versus `HTTPS` problem [https://stackoverflow.com/a/28385129].
